drivers/mil_passive_sonar/passive_sonar_driver/passive_sonar_driver.py

```diff
@@ -0,0 +1,590 @@
+#!/usr/bin/env python
+from __future__ import division
+
+import numpy as np
+
+import rospy
+import tf2_ros
+from tf import transformations
+from multilateration import Multilaterator, ls_line_intersection3d, get_time_delta
+
+import threading
+import serial
+import os
+
+from cv_bridge import CvBridge, CvBridgeError
+from sensor_msgs.msg import Image
+
+from visualization_msgs.msg import Marker
+from geometry_msgs.msg import Point
+from std_msgs.msg import Header
+from mil_passive_sonar.srv import *
+
+__author__ = 'David Soto'
+
+# GLOBALS
+lock = threading.Lock()  # prevent multiple threads simultaneously using a serial port
+
+def thread_lock(lock):  # decorator
```

Comment:
There's a `mil_tool` for this in `threading_helpers`. I think it does the same thing but it would reduce duplicated code.